### PR TITLE
added dispatch call for adding cab

### PR DESCRIPTION
**Pull request by Ayushi-knoldus**
What does this change do?
add a dispatch call to add Cab.

Any additional information for the reviewer to start
NA

How should this be manually tested?
We need the window OS in which Rust is installed. then execute this rust program

Which Trello task and sub-task does this change relate to?
CC - SP2 101.3 - Add dispatch call for add_cab

Are there any changes pending?
No

Does any team have to be notified of changes in this feature?
Yes

Definition of Done:

 - [ ] Is there >90% unit test code coverage?
        6.19%
 - [ ] Does this PR add new dependencies? If so, please list out the same.
 - [ ] Will this feature require a new piece of infrastructure to be implemented?
 - [x] Is there appropriate logging included?
 - [x] Does the project compile ok?
 - [x] Have Clippy violations been fixed?
 - [x] Have Code is properly formatted?

### CODE REVIEW

File: pallets/carpooling/src/lib.rs [hunk 1]
```diff
@@ -56,7 +56,7 @@ pub mod pallet {
     pub type Customer<T: Config> = StorageMap<_, Blake2_128Concat, T::AccountId, CustomerOf<T>>;
 	#[pallet::storage]
     #[pallet::getter(fn get_driver)]
-    pub type Driver<T: Config> = StorageMap<_, Blake2_128Concat, T::AccountId, DriverOf<T>>;
+    pub type Driver<T: Config> = StorageMap<_, Blake2_128Concat, u32, DriverOf<T>>;
 	#[pallet::storage]
     #[pallet::getter(fn get_booking)]
     pub type Booking<T: Config> = StorageMap<_, Blake2_128Concat, T::AccountId, T::AccountId>;
```
Comment (aman2909verma) [inline pallets/carpooling/src/lib.rs:62]: change AccountId to u32

**Review by aman2909verma — APPROVED** [review]

**Review by ankit-mogha — APPROVED** [review]

File: pallets/carpooling/src/lib.rs [hunk 2]
```diff
@@ -91,14 +91,23 @@ pub mod pallet {
 		/// An example dispatchable that takes a singles value as a parameter, writes the value to
 		/// storage and emits an event. This function must be dispatched by a signed extrinsic.
 		#[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
-		pub fn do_something(origin: OriginFor<T>, something: u32) -> DispatchResult {
+		pub fn add_new_cab(
```
Comment (bishtpawan) [inline pallets/carpooling/src/lib.rs:94]: please add documentation

File: pallets/carpooling/src/lib.rs [hunk 3]
```diff
@@ -91,14 +91,23 @@ pub mod pallet {
...
+			origin: OriginFor<T>,
+            cab_id: u32,
+            new_cab: DriverOf<T>,) -> DispatchResult {
 			// Check that the extrinsic was signed and get the signer.
 			// This function will return an error if the extrinsic is not signed.
 			// https://substrate.dev/docs/en/knowledgebase/runtime/origin
 			let who = ensure_signed(origin)?;
+			match <Driver<T>>::get(cab_id) {
```
Comment (bishtpawan) [inline pallets/carpooling/src/lib.rs:102]: try to update the flow

Comment (Ayushi-knoldus) [reply pallets/carpooling/src/lib.rs]: Done

File: pallets/carpooling/src/lib.rs (hunk 2 above)
Comment (Ayushi-knoldus) [reply pallets/carpooling/src/lib.rs]: Done

**Pull request closed without merging**